app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/CheckContactDetailsMrnController.scala

```diff
@@ -125,7 +125,12 @@ class CheckContactDetailsMrnController @Inject() (
                   .leftMap(err => Error(s"Could not remove contact details: ${err.message}"))
                   .fold(
                     e => logAndDisplayError("Submit Declarant Type error: ").apply(e),
-                    _ => Redirect(router.nextPageForChangeClaimantDetails(answer, featureSwitch))
+                    _ =>
+                      Redirect(
+                        router.CheckAnswers.when(fillingOutClaim.draftClaim.isComplete)(alternatively =
+                          router.nextPageForChangeClaimantDetails(No, featureSwitch)
```

Comment:
It just me, but I never liked Router hiding implementation details.
This `nextPageForChangeClaimantDetails(No, featureSwitch)` it almost like I need to go and check multiple places to understand complete picture.
I would certainly have it routing here. But it just mine opinion

Reply:
Agree. I'll rather move the `nextPageForChangeClaimantDetails` method into this class as it seems to be used only here anyway

Reply:
Thank you!